src/js/libgeo/GeoBasics.js

```diff
@@ -296,6 +312,19 @@ function addElementNoProof(el) {
         csgeo.ifs.push(el);
     }
 
+    // collect sets
+    var setsRe = new RegExp("^[P|L|S|C]s$"); // Ps, Ls, ...
```

Comment:
better use `var setsRe = /^[P|L|S|C]s$/;` for the constant RegExp here, because using the RegExp constructor function leads to possible slower runtime compilation of the regular expression. (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions )